python/mage/tgn/definitions/message_function.py

```diff
@@ -0,0 +1,46 @@
+import torch.nn as nn
+import torch
+
+
+class MessageFunction(nn.Module):
+    """
+    This is base class for Message function implementation
+    """
+
+    def __init__(
+        self, raw_message_dimension: int, message_dimension: int, device: torch.device
+    ):
+        super().__init__()
+        self.raw_message_dimension = raw_message_dimension
+        self.message_dimension = message_dimension
+        self.device = device
+
+
+class MessageFunctionMLP(MessageFunction):
+    def __init__(
+        self, raw_message_dimension: int, message_dimension: int, device: torch.device
+    ):
+        super().__init__(raw_message_dimension, message_dimension, device)
+
+        self.message_function_net = nn.Sequential(
+            nn.Linear(raw_message_dimension, raw_message_dimension // 2),
+            nn.ReLU(),
+            nn.Linear(raw_message_dimension // 2, message_dimension),
+        ).to(self.device)
+
+    def forward(self, data):
+        return self.message_function_net(data)
+
+
+class MessageFunctionIdentity(MessageFunction):
+    def __init__(
+        self, raw_message_dimension: int, message_dimension: int, device: torch.device
+    ):
+        super().__init__(raw_message_dimension, message_dimension, device)
+        assert raw_message_dimension == message_dimension, "Wrong!"
+
+    def forward(self, data):
```

Comment:
Actually, it shouldn't since this method is called part of `torch.nn.Module` as part of feed-forward neural network calculations. Here is a link to what was my [plan](https://pytorch.org/tutorials/beginner/pytorch_with_examples.html#pytorch-custom-nn-modules).  But I understand what you see as a problem since this method is not using any self variables. I would leave it like this now and maybe think about refactoring later.  @jmatak what do you think?